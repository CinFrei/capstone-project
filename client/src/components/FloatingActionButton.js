import { ReactComponent as AddIcon } from '../assets/add.svg'
import styled from 'styled-components/macro'

/** @component */
export default FloatingActionButton

function FloatingActionButton({ onClick }) {
  return (
    <ButtonStyled onClick={onClick}>
      <AddIcon />
    </ButtonStyled>
  )
}

const ButtonStyled = styled.button`
  background-color: var(--light-blue);
  border-radius: 30px;
  border: 2px solid var(--light-blue);
  bottom: 20px;
  display: grid;
  height: 60px;
  padding: 10px;
  place-items: center;
  position: fixed;
  right: 20px;
  width: 60px;
  z-index: 10;

  svg {
    margin: 0;
    width: 100%;
  }
  svg .st0 {
    fill: var(--primary-anthrazit);
  }
`
