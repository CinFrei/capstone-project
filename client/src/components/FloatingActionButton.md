Floating Action Button

```js static
<FloatingActionButton />
```
