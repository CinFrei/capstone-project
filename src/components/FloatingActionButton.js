import styled from 'styled-components/macro'
//import PropTypes from 'prop-types'

import add from '../assets/add.svg'

/** @component */
export default FloatingActionButton

function FloatingActionButton({ onClick }) {
  return (
    <ButtonStyled onClick={onClick}>
      <img src={add} alt="" />
    </ButtonStyled>
  )
}

const ButtonStyled = styled.button`
  position: fixed;
  bottom: 20px;
  right: 20px;
  z-index: 10;

  display: grid;
  place-items: center;
  width: 60px;
  height: 60px;
  padding: 10px;

  background-color: var(--light-blue);
  border: 2px solid var(--light-blue);
  border-radius: 30px;

  img {
    width: 100%;
    margin: 0;
  }
`
