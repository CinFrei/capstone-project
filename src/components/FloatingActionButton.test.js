import FloatingActionButton from './FloatingActionButton'
import { render } from '@testing-library/react'

describe('FloatingActionButton', () => {
  it('renders correctly', () => {
    const { container } = render(<FloatingActionButton />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
