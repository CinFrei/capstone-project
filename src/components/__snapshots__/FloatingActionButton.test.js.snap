// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FloatingActionButton renders correctly 1`] = `
.c0 {
  position: fixed;
  bottom: 20px;
  right: 20px;
  z-index: 10;
  display: grid;
  place-items: center;
  width: 60px;
  height: 60px;
  padding: 10px;
  background-color: var(--light-blue);
  border: 2px solid var(--light-blue);
  border-radius: 30px;
}

.c0 img {
  width: 100%;
  margin: 0;
}

<button
  class="c0"
>
  <img
    alt=""
    src="add.svg"
  />
</button>
`;
